### PR TITLE
Task/migrate to new blaze fetch campaigns endpoint

### LINKED ISSUE #10836: Stats widgets are showing "No network available" after some time
### Expected behavior
Stats widgets will refresh when the network is available.

### Actual behavior
Stats widgets, even with a constant WiFi and/or cellular connection, will randomly show "No network available".

image000000

### Steps to reproduce the behavior
I can't reliable reproduce it - just add a widget and let the phone sit locked for a while?

### PR DESCRIPTION
**Pull request by JorgeMucientes**
Closes: #10836 

### Description

Please don't panic with the number of file changes. Most of them are really small ones. 
This PR migrates existing endpoint for fetching Blaze campaigns to new endpoint. The previous one was deprecated and we were encouraged from the self serve team to migrate it. Since, the FluxC change were breaking changes I decided to help migrating the Jetpack/WordPress app at the same time as the WooCommerce app. I'd suggest checking the FluxC changes: https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2963 for extra context.

## To Test:

I'll appreciate some input here in case I'm missing something. But overall, the smoke testing Blaze campaign loading and opening campaign details should be enough. 

https://github.com/wordpress-mobile/WordPress-Android/assets/2663464/41e16b06-0cc0-4c7f-89ea-55f81b26b173

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WordPress/src/main/java/org/wordpress/android/ui/mysite/cards/blaze/CampaignStatus.kt [hunk 1]
```diff
@@ -8,12 +8,12 @@ import org.wordpress.android.util.AppLog
...
 enum class CampaignStatus(val status: String, @StringRes val stringResource: Int) {
+    InModeration("pending", R.string.campaign_status_in_moderation),
+    Scheduled("scheduled", R.string.campaign_status_scheduled),
     Active("active", R.string.campaign_status_active),
-    Completed("finished", R.string.campaign_status_completed),
     Rejected("rejected", R.string.campaign_status_rejected),
     Canceled("canceled", R.string.campaign_status_canceled),
-    Scheduled("scheduled", R.string.campaign_status_scheduled),
-    InModeration("created", R.string.campaign_status_in_moderation);
+    Completed("finished", R.string.campaign_status_completed);
```
Comment (JorgeMucientes) [inline WordPress/src/main/java/org/wordpress/android/ui/mysite/cards/blaze/CampaignStatus.kt:16]: The new API exposes the following statues: 

Screenshot 2024-03-04 at 17 37 02

The only change to what we were using before is that now `created` value has been replaced with `pending`. The rest of the returned values from the API are the same, I've just simply ordered them in "successive campaign life cycle" order.

**Review by AjeshRPai — COMMENTED** [review]
Hey @JorgeMucientes 

Thanks for picking up on this change. I smoke-tested the app and found a minor issue. When you go to the campaign listing screen, the first item is the most recent campaign, but if you refresh, it is not arranged in the most recent order. See the video capture below. 

[Screen_recording_20240306_114536.webm](https://github.com/wordpress-mobile/WordPress-Android/assets/17463767/0cfc8bd2-d94b-456b-8770-a6cabdb1716d)

I haven't reviewed the code yet. I plan to do that today or tomorrow. But I wanted to let you know about this issue so that I can possibly take up the review after it has been fixed.

**Comment by JorgeMucientes** [comment]
Hey @AjeshRPai thanks for the input! I've checked the issue you mentioned and I noticed that I can only reproduce it on `trunk`. I can't reproduce it on ` task/migrate-to-new-blaze-fetch-campaigns-endpoint`. This are the screen recordings from my tests: 

`trunk`:

https://github.com/wordpress-mobile/WordPress-Android/assets/2663464/0735c99c-6940-4c5c-a17b-063d3c58a586

` task/migrate-to-new-blaze-fetch-campaigns-endpoint`:


https://github.com/wordpress-mobile/WordPress-Android/assets/2663464/95aa9269-2a15-4fd2-9285-995f563e61e0


Upon further investigation I noticed that the issue is with the API response. Its returning the items for `page=1` in different orders. 

So once we merge this PR that issue should be fixed as it seems is not happening with the new endpoint.

**Review by AjeshRPai — APPROVED** [review]
Hey @JorgeMucientes 

I have smoke tested the blaze feature and everything looks good to me. Thanks for wrangling this change. I will approve the fluxC PR. Let me know if you need me to update the fluxC hash version and merge this.

**Pull request merged**